### PR TITLE
FIX: lm pre-training

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: finetune/base.py [hunk 1]
```diff
@@ -514,12 +520,12 @@ def _define_placeholders(self):
...
         self.M = tf.placeholder(tf.float32, [None, self.config.max_length])  # sequence mask
         # when target dim is not set, an array of [None] targets is passed as a placeholder
-        self.Y = tf.stop_gradient(tf.placeholder(tf.float32, [None, self.target_dim or 1]))  # classification targets
+
         self.do_dropout = tf.placeholder(tf.float32)  # 1 for do dropout and 0 to not do dropout
         if self.target_type == SEQUENCE_LABELING:
             self.Y = tf.placeholder(tf.int32, [None, self.config.max_length])  # classification targets
         else:
-            self.Y = tf.placeholder(tf.float32, [None, self.target_dim])  # classification targets
+            self.Y = tf.stop_gradient(tf.placeholder(tf.float32, [None, self.target_dim or 1]))  # classification targets
```
Comment (benleetownsend) [inline finetune/base.py]: Gradients already do not flow into placeholders The `tf.stop_gradient` should be on the loss function to protect against changes to `self.Y`

File: finetune/utils.py [hunk 2]
```diff
@@ -95,6 +95,14 @@ def np_init(w):
     return partial(_np_init, w=w)
 
 
+def guarantee_initialized_variables(sess):
```
Comment (benleetownsend) [inline finetune/utils.py]: I really like this.

File: finetune/utils.py [hunk 3]
```diff
@@ -95,6 +95,14 @@ def np_init(w):
...
+    global_vars = tf.global_variables()
```
Comment (madisonmay) [inline finetune/utils.py]: Credit stackoverflow

File: finetune/base.py [hunk 4]
```diff
@@ -489,6 +493,8 @@ def _build_model(self, n_updates_total, target_dim, train=True):
         elif not self.is_trained:
             self._load_base_model()
 
+        guarantee_initialized_variables(self.sess)
```
[the remark this answers is not in the document]
Comment (madisonmay) [reply finetune/base.py]: Specify model/clf as scope
